src/OrbitGl/OpenGlTextRenderer.cpp

```diff
@@ -335,9 +335,14 @@ void OpenGlTextRenderer::AddText(const char* text, float x, float y, float z,
     return;
   }
 
+  float string_width = GetStringWidth(text, formatting.font_size);
+  string_width = std::min(string_width, formatting.max_size > 0 ? formatting.max_size : FLT_MAX);
+
   if (formatting.halign == HAlign::Right) {
+    pen_.x -= string_width;
+  } else if (formatting.halign == HAlign::Centered) {
     float string_width = GetStringWidth(text, formatting.font_size);
```

Comment:
shouldn't we use the same string_width in both cases?

Reply:
D'oh, missed this. Yes. Thanks for catching!